definitions/procedures/backup/prepare_directory.rb

```diff
@@ -22,8 +23,14 @@ def run
 
       FileUtils.rm(Dir.glob(File.join(@backup_dir, '.*.snar'))) if @preserve_dir
       if @incremental_dir
-        FileUtils.cp(Dir.glob(File.join(@incremental_dir, '.*.snar')), @backup_dir)
+        snar_files = Dir.glob(File.join(@incremental_dir, '.*.snar'))
+        if snar_files.empty?
```

Comment:
You can combine `!snar_files.empty?` condition with parent if condition ?  As we should take incremental backup only when snar_files are available?

Reply:
@upadhyeammit 
In `snar_files = Dir.glob(File.join(@incremental_dir, '.*.snar'))` we use `@incremental_dir`. So I think it is good to first check if `@incremental_dir` exist. If we shift the `!snar_files.empty?` to parent then foreman-maintain will every time try to check the snar files without checking if  `@incremental_dir` exist. 
Let me know your thoughts on above.

Reply:
Agree with you !